gensim/models/word2vec.py

```diff
@@ -1610,13 +1612,16 @@ def accuracy(self, questions, restrict_vocab=30000, most_similar=most_similar, u
                     logger.debug("skipping line #%i with OOV words: %s" % (line_no, line.strip()))
                     continue
 
+                original_vocab = self.vocab
+                self.vocab = ok_vocab
                 ignore = set(self.vocab[v].index for v in [a, b, c])  # indexes of words to ignore
                 predicted = None
                 # find the most likely prediction, ignoring OOV words and input words
                 sims = most_similar(self, positive=[b, c], negative=[a], topn=False, restrict_vocab=restrict_vocab)
```

Comment:
The behaviour for `restrict_vocab` in `most_similar` is a little different from its behaviour in `accuracy`
In accuracy, it simply takes the first `restrict_vocab` vectors for calculating similarity, and in `accuracy`, `ok_vocab` is created by sorting by frequency and taking first `restrict_vocab` words.

This could cause some unintended behaviour, especially with low values for `restrict_vocab` since it is passed to `most_similar` from accuracy. The two sets of vocab used in the two methods could be significantly different, and possibly even completely disjoint.

Setting `self.vocab = ok_vocab` before the `most_similar` call should handle this, I think?